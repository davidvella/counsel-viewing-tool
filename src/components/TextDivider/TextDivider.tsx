import withStyles from '@material-ui/core/styles/withStyles';
import createStyles from '@material-ui/core/styles/createStyles';
import Typography from '@material-ui/core/Typography';
import React from "react";
import { WithStyles } from '@material-ui/styles/withStyles';
import Divider from '@material-ui/core/Divider';

const styles = (theme: any) => createStyles({
  container: {
    height: "40px",
    "line-height": "40px"
  },
  label: {
    float: "left",
    "padding-right": "5px",
    "line-height": "40px"
  },
  divider: {
    "margin-bottom": "10px",
    "padding-top": "19px",
    "line-height": "40px"
  }
});

export interface ITextDividerProps extends WithStyles<typeof styles> {
  /**
   * The Text to be displayed.
   */
  label?: React.ReactNode;
  /**
   * If true, the divider will have a lighter color.
   */
  light?: boolean;
}

/*************************************************************************************
 * This is an extension to the material design divider which allows text to be added.
 *************************************************************************************/
export const TextDivider = withStyles(styles)(
  class extends React.Component<ITextDividerProps, any> {

    /**
      * Default React component render method
      */
    public render() {
      const { classes } = this.props;

      return (
        <div className={classes.container}>
          <Typography variant="body2" className={classes.label}>
            {
              this.props.label
            }
          </Typography>
          <div className={classes.divider}>
            <Divider light={this.props.light} />
          </div>
        </div>
      );
    }
  }
);

