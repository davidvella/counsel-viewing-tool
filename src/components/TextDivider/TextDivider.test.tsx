import React from 'react';
import {
    render,
    cleanup,
} from "@testing-library/react";
// this adds custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
import { TextDivider } from './TextDivider';

describe('<Text Divider />', () => {

    afterEach(cleanup);

    it("renders the component", () => {
        const { container } = render(<TextDivider label="Flags" />);
        expect(container.firstChild).toMatchSnapshot();
    });
});