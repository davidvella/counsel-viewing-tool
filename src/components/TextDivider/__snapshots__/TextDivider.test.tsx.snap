// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text Divider /> renders the component 1`] = `
<div
  class="Component-container-1"
>
  <p
    class="MuiTypography-root Component-label-2 MuiTypography-body2"
  >
    Flags
  </p>
  <div
    class="Component-divider-3"
  >
    <hr
      class="MuiDivider-root"
    />
  </div>
</div>
`;
